### ISSUE TITLE
Black window of roboception cloud viewer and Error of compling vtk

### ISSUE THREAD

**Issue by hanlinniu** [roboception/rc_cloud_accumulator#3]
After I run rosrun rc_cloud_accumulator rc_cloud_accumulator, I got black window of roboception cloud viewer. No matter how I drag and click the mouse, it is still black. By the way, the service of "/rc_cloud_accumulator/save_cloud" works well, I can save the view the point cloud pcd file. 

When I catkin_make the src files of rc_cloud_accumulator, I found I had the following warnings (don't know if it is the problem to cause the black window):

The imported target "vtkRenderingPythonTkWidgets" references the file
   "/usr/lib/x86_64-linux-gnu/libvtkRenderingPythonTkWidgets.so"
but this file does not exist.  Possible reasons include:
* The file was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and contained
   "/usr/lib/cmake/vtk-6.2/VTKTargets.cmake"
but not all the files it references.

The imported target "vtk" references the file
   "/usr/bin/vtk"
but this file does not exist.  Possible reasons include:
* The file was deleted, renamed, or moved to another location.
* An install or uninstall procedure did not complete successfully.
* The installation package was faulty and contained
   "/usr/lib/cmake/vtk-6.2/VTKTargets.cmake"
but not all the files it references.

**Comment by flixr** [comment]
Seems to be some problem with the VTK package you have installed...
Did you install all dependencies via `rosdep install --from-paths . --ignore-src -r -y`?

This package is also already available pre-built from the ROS buildfarm, just `apt install ros-$ROS_DISTRO-rc-cloud-accumulator`.

**Comment by hanlinniu** [comment]
The commands I used are: 

1, roscore

2, rosrun rc_visard_driver rc_visard_driver _device:=:XXXXXX  _enable_tf:=True _autostart_dynamics:=True _autostop_dynamics:=True _ptp_enabled:=True _autostart_dynamics_with_slam:=True _autopublish_trajectory:=True 

3, rosrun rc_cloud_accumulator rc_cloud_accumulator

**Comment by hanlinniu** [comment]
> Seems to be some problem with the VTK package you have installed...
> Did you install all dependencies via `rosdep install --from-paths . --ignore-src -r -y`?
> 
> This package is also already available pre-built from the ROS buildfarm, just `apt install ros-$ROS_DISTRO-rc-cloud-accumulator`.

I installed all dependencies and also tried pre-built ros-kinetic-rc-cloud-accumulator, still got black window. Do you think it is because I am using PCL 1.7 and also ros-pcl? Does PCL 1.7 conflict with ros-pcl?

**Comment by hanlinniu** [comment]
> Seems to be some problem with the VTK package you have installed...
> Did you install all dependencies via `rosdep install --from-paths . --ignore-src -r -y`?
> 
> This package is also already available pre-built from the ROS buildfarm, just `apt install ros-$ROS_DISTRO-rc-cloud-accumulator`.

I tried on a new laptop, installed ros-kinetic-rc-cloud-accumulator, still got black window. This new laptop also has PCL 1.7 and ros-kinetic-pcl-ros installed.

**Comment by flixr** [comment]
I don't see how it could be related to PCL, since it seems to only not show the pointcloud, but receive it and save it just fine (which you should see in the console output). You can open the saved pcd with pcl_viewer?
On Ubuntu 18.04/melodic it also runs without problems using PCL 1.7.

> Does PCL 1.7 conflict with ros-pcl?

You mean pcl_ros? Not that I know of... that just contains bindings.

**Comment by hanlinniu** [comment]
> I don't see how it could be related to PCL, since it seems to only not show the pointcloud, but receive it and save it just fine (which you should see in the console output). You can open the saved pcd with pcl_viewer?
> On Ubuntu 18.04/melodic it also runs without problems using PCL 1.7.
> 
> > Does PCL 1.7 conflict with ros-pcl?
> 
> You mean pcl_ros? Not that I know of... that just contains bindings.

Yes, I can save the merged point cloud as a pcd file and pcl_viewer it. but cannot view it using roboception cloud viewer. I am using Ubuntu 16.04 and installed ros-kinetic-pcl-ros together with ros-kinetic-desktop-full

**Comment by flixr** [comment]
Sorry, I currently have no idea what the problem could be and I cannot reproduce it here.
After moving the view, I can see the pointcloud as expected.

**Comment by hanlinniu** [comment]
> Sorry, I currently have no idea what the problem could be and I cannot reproduce it here.
> After moving the view, I can see the pointcloud as expected.

I just found that when I rostopic echo /trajectory, there is no values. But I got values from rostopic echo /pose and rostopic echo /stereo/points2, do you think I had problem of publishing /trajectory? The command I am using is: 

1, roscore

2, rosrun rc_visard_driver rc_visard_driver _device:=:XXXXXX _enable_tf:=True _autostart_dynamics:=True _autostop_dynamics:=True _ptp_enabled:=True _autostart_dynamics_with_slam:=True _autopublish_trajectory:=True

3, rosrun rc_cloud_accumulator rc_cloud_accumulator

Are they right?

**Comment by flixr** [comment]
It has nothing to do with that, the complete trajectory is only generated upon request when you call `get_trajectory`, see https://github.com/roboception/rc_cloud_accumulator/blob/master/README.md#what-it-does

Getting the trajectory will re-orient the individual views (e.g. after a loop closure), in the mean time the live pose is used.

**Comment by hanlinniu** [comment]
> It has nothing to do with that, the complete trajectory is only generated upon request when you call `get_trajectory`, see https://github.com/roboception/rc_cloud_accumulator/blob/master/README.md#what-it-does
> 
> Getting the trajectory will re-orient the individual views (e.g. after a loop closure), in the mean time the live pose is used.

Ok, this is my terminal messages, don't know if it is helpful or not:

:~$ rosrun rc_cloud_accumulator rc_cloud_accumulator
[ INFO] [1593016826.197885918]: Distance filter range: 0.00m - 5.00m
[ INFO] [1593016826.197952490]: Voxel grid filter for display: 0.050m
[ INFO] [1593016826.197965963]: Voxel grid filter for saving:  0.010m
[ INFO] [1593016826.500328068]: Received first live pose
[ INFO] [1593016827.568838699]: Received 1 point cloud
[ INFO] [1593016827.571652844]: Change view to start displaying the clouds.
[ INFO] [1593016837.663417262]: Received 30 point clouds
[ INFO] [1593016847.709367420]: Received 59 point clouds
[ INFO] [1593016858.012123280]: Received 89 point clouds
[ INFO] [1593016868.290844460]: Received 118 point clouds

**Comment by flixr** [comment]
If there are no VTK errors, this unfortunately does not help.
They are received and processed, just not rendered by VTK... no idea why... works here. Sorry.

**Comment by flixr** [comment]
You could try installing ros-kinetic-rc-cloud-accumulator in the ros kinetic (or melodic) docker image and run it this way...
See also http://wiki.ros.org/docker/Tutorials/GUI
